### DISCUSSION TITLE
dynamically updating pie chart options is causing an error

### DISCUSSION THREAD

**Discussion by AmosSzust** [apexcharts/apexcharts.js#4860] [Q&A]
This happens with apexcharts 4.2.0, it doesn't happen with 4.1.0
```
chartOptions.value = {
            ...chartOptions.value,
            ...{
              labels: response.data.map(
                (sector: theModel) => sector.sector,
              ),
            }
          };
```
Should render the chart with the relevant data and labels
chart is not shown in the console I see Cannot read properties of undefined (reading '0')
it is related to an array but not sure which array exactly, both labels and series have values.
when downgrading to apexcharts 4.1.0, it works.
I use "vue3-apexcharts": "1.8.0"

**Comment by LentoMan** [comment]
We are not using vue, but [this change](https://github.com/apexcharts/apexcharts.js/commit/a25c98bc6d1c0a2168782329b4b37b4f7e0551c5#diff-c799981553c8e8e3e5321e944950ffef1fc4776a77b12bc138be46325bda1543R424) seems to break things for us with 4.2.0 when we only trying to update a small part of the config, perhaps it could be related?

**Reply by AmosSzust** [reply]
Maybe related, the issue is when trying to make any kind of change in the options of the chart.

**Comment by junedchhipa** [comment]
I think, the line this.opts is the culprit here
I will remove that line and release a new version soon.
